python/src/aiconfig/editor/client/src/components/prompt/PromptContainer.tsx

```diff
@@ -75,6 +82,17 @@ export default memo(function PromptContainer({
           prompt={prompt}
           promptSchema={promptSchema}
           onUpdateModelSettings={updateModelSettings}
+          onUpdateParameters={(data) => {
+            const newParameters: Record<string, unknown> = {};
+            for (const paramTuple of data.newParameters ?? []) {
+              const key = paramTuple.parameterName;
+              const val = paramTuple.parameterValue;
+
+              newParameters[key] = val;
+            }
+
+            onUpdateParameters(index, newParameters);
+          }}
```

Comment:
Should move this logic up into the `useCallback`, otherwise this inline function just makes the above memoization wasted